esp-hal/src/timer/systimer.rs

```diff
@@ -332,54 +264,26 @@ pub trait Unit {
     }
 }
 
-/// A specific [Unit]. i.e. Either unit 0 or unit 1.
-#[derive(Debug)]
-pub struct SpecificUnit<'d, const CHANNEL: u8>(PhantomData<&'d ()>);
-
-impl<const CHANNEL: u8> SpecificUnit<'_, CHANNEL> {
-    fn new() -> Self {
-        Self(PhantomData)
-    }
-}
-
-impl<const CHANNEL: u8> Unit for SpecificUnit<'_, CHANNEL> {
-    fn channel(&self) -> u8 {
-        CHANNEL
-    }
-}
-
-/// Any [Unit]. Could be either unit 0 or unit 1.
+/// An alarm unit
 #[derive(Debug)]
-pub struct AnyUnit<'d>(PhantomData<&'d ()>, u8);
-
-impl Unit for AnyUnit<'_> {
-    fn channel(&self) -> u8 {
-        self.1
-    }
-}
-
-impl<'d, const CHANNEL: u8> From<SpecificUnit<'d, CHANNEL>> for AnyUnit<'d> {
-    fn from(_value: SpecificUnit<'d, CHANNEL>) -> Self {
-        Self(PhantomData, CHANNEL)
-    }
+pub struct Alarm {
+    comp: u8,
+    unit: Unit,
 }
 
-impl<'d, const CHANNEL: u8> TryFrom<AnyUnit<'d>> for SpecificUnit<'d, CHANNEL> {
-    type Error = u8;
-
-    fn try_from(value: AnyUnit<'d>) -> Result<Self, Self::Error> {
-        if value.1 == CHANNEL {
-            Ok(SpecificUnit::new())
-        } else {
-            Err(value.1)
+impl Alarm {
+    const fn new(comp: u8) -> Self {
+        Alarm {
+            comp,
+            unit: Unit::Unit0,
         }
     }
-}
 
-/// A comparator that can generate alarms/interrupts based on values of a unit.
-pub trait Comparator {
     /// Returns the comparators number.
```

Comment:
```suggestion
    /// Returns the comparator's number.
```

I know it's preexisting, but we can fix it 🙃